src/zope/password/password.py

```diff
@@ -492,14 +491,13 @@ def checkPassword(self, encoded_password, password):
         return encoded_password == self.encodePassword(password)[5:]
 
     def match(self, encoded_password):
-        if not isinstance(encoded_password, bytes):
-            encoded_password = encoded_password.encode('ascii')
+        encoded_password = _encoder(encoded_password)
         return (
-            encoded_password.startswith(b'{SHA}') or
+            encoded_password.startswith(self._prefix) or
             encoded_password.startswith(b'{SHA1}'))
```

Comment:
You could shorten this to a single `.startswith((self._prefix, b'{SHA1}'))` call.

Reply:
Thanks. I always forget that. Possibly because if you leave off the extra set of parentheses, bad things happen.